app/controllers/api/v1/classifications_controller.rb

```diff
@@ -36,10 +36,23 @@ def build_resource_for_update(update_params)
     classification
   end
 
+  def sms_ids_from_links(link_params)
+    sms = if ids = link_params.delete(:set_member_subjects)
+      SetMemberSubject.select(:id).find(ids)
```

Comment:
We need to raise a not found error if a subject id is sent that does match
the database. I can pluck and then test the result against the number we
were expecting. I was just using find since it did that automatically.
On Jan 29, 2015 8:23 AM, "Campbell Allen" <notifications@github.com> wrote:

> In app/controllers/api/v1/classifications_controller.rb
> <https://github.com/zooniverse/Panoptes/pull/458#discussion_r23769950>:
>
> > @@ -36,10 +36,23 @@ def build_resource_for_update(update_params)
> >      classification
> >    end
> >
> > +  def sms_ids_from_links(link_params)
> > +    sms = if ids = link_params.delete(:set_member_subjects)
> > +      SetMemberSubject.select(:id).find(ids)
>
> likewise here SetMemberSubject.where(id: ids).pluck(:id)
> then you wont have to try the array on line 48.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/458/files#r23769950>.
>